unit_tests/test_save_data/test_save_players.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+from database.postgres import save_api_data
+
+class TestPlayersSavingData:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.sport_name = "mma"
+        self.json_case_empty = {}
+        self.json_case_empty_parameters = {
+            "get": "players",
+            'parameters': {},
+            "response": {
+                "player": {
+                    "id": 276,
+                    "name": "Neymar",
+                    "photo": "https://media.api-sports.io/football/players/276.png"
+                }
+            }
+        }
+        self.json_case_valid_parameters = {
+            "get": "players",
+            'parameters': {
+                'team': 1488
+            }
+        }
+        self.json_case_full_bad_info = {
+            "get": "players",
+            'parameters': {},
+            "response": [
+                {
+                    "player": {
+                        "id": 276,
+                        "name": "Neymar",
+                        "photo": "https://media.api-sports.io/football/players/276.png",
+                        "team": {
+                            "id": 40,
+                            "name": "SBG Ireland"
+                        }
+                    }
+                },
+                {
+                    "player": {
+                        "id": 301,
+                        "name": "John Doe",
+                        "photo": "https://example.com/john_doe.png",
+                        "team": {
+                            "id": 50,
+                            "name": "Team Alpha"
+                        }
+                    }
+                }
+            ]
+        }
+        self.json_case_full_info = {
+            "get": "players",
+            'parameters': {},
+            "response": [
+                {
+                    "id": 276,
+                    "name": "Neymar",
+                    "photo": "https://media.api-sports.io/football/players/276.png",
+                    "team": [
+                        {
+                            "id": 40,
+                            "name": "SBG Ireland"
+                        }
+                    ]
+                },
+                {
+                    "id": 302,
+                    "name": "John Doe",
+                    "photo": "https://example.com/john_doe.png",
+                    "team": [
+                        {
+                            "id": 55,
+                            "name": "Beta Warriors"
+                        }
+                    ]
+                }
+            ]
+        }
+
+    def test_parameters_missing(self):
+        save_api_data(self.json_case_empty, self.sport_name)
```

Comment:
This is Integration Testing which will impact the database.
What I need is UnitTesting.

For example:
You test some Service (let's say UserService)

That service has its methods (e.g. create_user). Those methods, however depend on different DALs and other stuff.

Your job is to create Mocks of those dependencies. For example -> fake dal that will not actually call the database, instead you will just tell it "hey, when I give you this set of data, you return this stuff, ok?". And then you just test the "create_user" method itself. We can talk more on it during the meeting, ot you can just watch some videos on YouTube, because this one is a little to complicated for me to explain in a single GitHub comment